src/webgpu/shader/validation/expression/call/builtin/all.spec.ts

```diff
@@ -0,0 +1,173 @@
+const builtin = 'all';
+export const description = `
+Validation tests for the ${builtin}() builtin.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf, objectsToRecord } from '../../../../../../common/util/data_tables.js';
+import { Type, elementTypeOf, kAllScalarsAndVectors } from '../../../../../util/conversion.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+import { validateConstOrOverrideBuiltinEval } from './const_override_validation.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kArgumentTypes = objectsToRecord(kAllScalarsAndVectors);
+
+g.test('argument_types')
+  .desc(
+    `
+Validates that scalar and vector arguments are rejected by ${builtin}() if not bool or vecN<bool>
+`
+  )
+  .params(u => u.combine('type', keysOf(kArgumentTypes)))
+  .fn(t => {
+    const type = kArgumentTypes[t.params.type];
+    validateConstOrOverrideBuiltinEval(
+      t,
+      builtin,
+      /* expectedResult */ elementTypeOf(type) === Type.bool,
+      [type.create(0)],
+      'constant',
+      /* returnType */ Type.bool
+    );
+  });
+
+const kTests = {
+  valid: {
+    src: `_ = ${builtin}(true);`,
+    pass: true,
+  },
+  alias: {
+    src: `_ = ${builtin}(bool_alias(true));`,
+    pass: true,
+  },
+  bool: {
+    src: `_ = ${builtin}(false);`,
+    pass: true,
+  },
+  i32: {
+    src: `_ = ${builtin}(1i);`,
+    pass: false,
+  },
+  u32: {
+    src: `_ = ${builtin}(1u);`,
+    pass: false,
+  },
+  f32: {
+    src: `_ = ${builtin}(1.0f);`,
+    pass: false,
+  },
+  f16: {
+    src: `_ = ${builtin}(1.0h);`,
+    pass: false,
+  },
+  vec_bool: {
+    src: `_ = ${builtin}(vec2<bool>(false, true));`,
```

Comment:
Please can we also have a `vec2(false, true)`, `vec3(true)` ?

Reply:
Done.